cmd/zed/agents/zfs_mod.c

```diff
@@ -1279,6 +1290,41 @@ zfs_enum_pools(void *arg)
 	return (NULL);
 }
 
+#define	CONFIG_LINE_MAX 100
+
+static int
+zed_lookup_tunable(const char *zedlet_dir, const char *tuneable,
```

Comment:
`zed_lookup_tunable()` is basically a lightweight shell parser.  It's going to get tripped up if you do things like `ZED_DISK_LABEL_WAIT =    1` or `ZED_DISK_LABEL_WAIT=1` or:
```sh
if dmidecode | grep "system-that-needs-more-spin-up-time" ; then
	ZED_DISK_LABEL_WAIT = 6000
else
	ZED_DISK_LABEL_WAIT = 3000
fi
```
I was thinking you could just "execute" zed.rc and then print out all its variables.  Or more specifically, have zed call `libzfs_run_process_get_stdout_nopath()` with `bash -c 'set -o allexport && source zed.rc && env'` as the command.  That function will return an array of lines of output which you can extract the `tuneable` from.  You can use `vdev_run_cmd()` as a template for how to use `libzfs_run_process_get_stdout_nopath()` since it's a little complicated.

I am happy that with this PR we would now have a true zed-readable config file (not just for zedlets)!  That's something we've wanted for a long time.

Reply:
The `sscanf` does correctly process whitespace on either side of the '='.  I tested with many variations, including tab characters.

The format of zed.rc is only implied (name=value) but not otherwise defined AFAIKT.  If it is intended to be a bash script then I'm more inclined to used a different config file for the non-zedlet tunables. I'd like to keep it simple and useable for other future non-zedlet tunables.

Reply:
Ok, that's good that it's interpreting white-space correctly. 

> The format of zed.rc is only implied (name=value) but not otherwise defined AFAIKT. 

The `rc` (run command) extension typically implies the file is executed as configuration commands.  Since we're already executing `zed.rc` at the beginning of our zedlets, I think it's fair for users to assume it's a script and that they could put other shell commands in there.

I think you're right that a separate zed daemon config file might be the way to go.